compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -2457,6 +2459,7 @@ object SymDenotations {
       !owner.isClass
       || owner.isRefinementClass
       || owner.is(Scala2x)
+      || denot.is(Synthetic)
```

Comment:
I don't see how this can be justified. Incremental compilation might have defined a synthetic in run 1 and then stopped defining the synthetic in run 2. So the synthetic is no longer valid (does not exist).

Reply:
fully this should be:
```
(denot.is(Synthetic) && denot.is(ModuleClass) && stillValidInOwner(denot.companionClass))
```
this will cath our case and stillValidInOwner(denot.companionClass)  will sure that companiounClass exists.

(will move to a separate patch now,  since it's more simple than ownity).

Reply:
(now in #11400 )